modules/util/css-builder/src/main/java/com/liferay/css/builder/CSSBuilder.java

```diff
@@ -58,67 +65,56 @@
  * @author Shuyang Zhou
  * @author David Truong
  */
-public class CSSBuilder implements AutoCloseable {
+public class CSSBuilder implements AutoCloseable, IDefaultProvider {
 
 	public static void main(String[] args) throws Exception {
-		Map<String, String> arguments = ArgumentsUtil.parseArguments(args);
+		Map<String, String> argMap = new HashMap<>();
 
-		List<String> dirNames = new ArrayList<>();
+		for (int x = 0; x < args.length; x++) {
+			if (args[x].indexOf('=') != args[x].length() - 1) {
+				String[] arg = args[x].replace('=', ' ').split(" ");
 
-		String dirName = GetterUtil.getString(
-			arguments.get("sass.dir"), CSSBuilderArgs.DIR_NAME);
+				if (arg.length == 2) {
+					if (arg[0].startsWith("sass.dir.")) {
+						arg[0] = "sass.dir";
+					}
 
-		if (Validator.isNotNull(dirName)) {
-			dirNames.add(dirName);
-		}
-		else {
-			for (int i = 0;; i++) {
-				dirName = arguments.get("sass.dir." + i);
-
-				if (Validator.isNotNull(dirName)) {
-					dirNames.add(dirName);
-				}
-				else {
-					break;
+					argMap.put(arg[0], arg[1]);
 				}
 			}
 		}
 
-		boolean appendCssImportTimestamps = GetterUtil.getBoolean(
-			arguments.get("sass.append.css.import.timestamps"),
-			CSSBuilderArgs.APPEND_CSS_IMPORT_TIMESTAMPS);
-		String docrootDirName = GetterUtil.getString(
-			arguments.get("sass.docroot.dir"), CSSBuilderArgs.DOCROOT_DIR_NAME);
-		boolean generateSourceMap = GetterUtil.getBoolean(
-			arguments.get("sass.generate.source.map"));
-		String outputDirName = GetterUtil.getString(
-			arguments.get("sass.output.dir"), CSSBuilderArgs.OUTPUT_DIR_NAME);
+		Collection<String> argsCollection = new ArrayList<>();
 
-		String portalCommonPath = arguments.get("sass.portal.common.path");
-
-		if (Validator.isNull(portalCommonPath)) {
-			portalCommonPath = arguments.get("sass.portal.common.dir");
+		for (Entry<String, String> entry : argMap.entrySet()) {
+			argsCollection.add(entry.getKey());
+			argsCollection.add(entry.getValue());
 		}
 
-		int precision = GetterUtil.getInteger(
-			arguments.get("sass.precision"), CSSBuilderArgs.PRECISION);
-		String[] rtlExcludedPathRegexps = StringUtil.split(
-			arguments.get("sass.rtl.excluded.path.regexps"));
-		String sassCompilerClassName = arguments.get(
-			"sass.compiler.class.name");
+		String[] argsArray = argsCollection.toArray(new String[0]);
+
+		try (CSSBuilder cssBuilder = new CSSBuilder()) {
+			Builder builder = JCommander.newBuilder();
 
-		try (CSSBuilder cssBuilder = new CSSBuilder(
-				appendCssImportTimestamps, docrootDirName, generateSourceMap,
-				outputDirName, portalCommonPath, precision,
-				rtlExcludedPathRegexps, sassCompilerClassName)) {
+			JCommander commander = builder.addObject(
```

Comment:
Please avoid chaining. I think streams is the only fluent interface that Brian accepts.